src/Input/Select/Select.tsx

```diff
@@ -24,82 +24,118 @@ const Select: ISelect = (props: Props) => {
     onFocus,
     onBlur,
     onChange,
+    onInputChange,
     noOptionResult = 'No results found',
     small,
     disableTyping,
     removeFloatingLabel,
     removeDropIcon,
     error,
     renderError,
-    value, // eslint-disable-line @typescript-eslint/no-unused-vars
-    children, // eslint-disable-line @typescript-eslint/no-unused-vars
+    value,
+    defaultValue,
+    children,
     isLoading,
     ...defaultProps
   } = props;
 
+  // warn deprecated prop "status"
+  React.useEffect(
+    function warnDeprecatedProp() {
+      if (status) {
+        if (typeof console !== 'undefined') {
+          console.warn(`Warning: Select's status prop is deprecated and will be
+        removed in a future release.\n\nPlease use the error prop instead to
+        show errors and indicate an error state.`);
+        }
+      }
+    },
+    [status]
+  );
+
   const [floating, setFloating] = React.useState<boolean>(false);
   const [isFocus, setIsFocus] = React.useState<boolean>(false);
-  const [selectedValue, setSelectedValue] = React.useState<string>('');
-  const [filterValue, setFilterValue] = React.useState<React.ReactNode[]>([]);
-  const [cursor, setCursor] = React.useState<number>(0);
-  const [shouldScrollToCursor, setShouldScrollToCursor] = React.useState<
-    boolean
-  >(false);
-  const [defaultValue, setDefaultValue] = React.useState<string>(
-    props.defaultValue
+  const [inputValue, setInputValue] = React.useState<string>(
+    value || defaultValue || ''
   );
-
-  const node: React.RefObject<HTMLDivElement> = React.useRef();
-
-  const handleOnBlur = React.useCallback((event: MouseEvent) => {
-    const element = event.target as HTMLElement;
-    if (!ReactDOM.findDOMNode(node.current).contains(element)) {
-      setIsFocus(false);
+  const [activeOptionIndex, setActiveOptionIndex] = React.useState<number>(0);
+  const [
+    shouldScrollToActiveOption,
+    setShouldScrollToActiveOption,
+  ] = React.useState<boolean>(false);
+
+  const selectContainerRef: React.RefObject<HTMLDivElement> = React.useRef();
+
+  // set options based on children and inputValue
+  const getAvailableOptions = React.useCallback(() => {
```

Comment:
Do you think it is possible to use `const availabeOptions = useMemo(// ...)` to instead of `getAvailableOptions` and `const [options, setOptions] = React.useState(//...)`?